src/Common/Core/Impl/Threading/MainThreadExtensions.cs

```diff
@@ -12,6 +14,14 @@ public static MainThreadAwaitable SwitchToAsync(this IMainThread mainThread, Can
         public static bool CheckAccess(this IMainThread mainThread)
             => Thread.CurrentThread.ManagedThreadId == mainThread.ThreadId;
 
+
+        [Conditional("TRACE")]
+        public static void Assert(this IMainThread mainThread, [CallerMemberName] string memberName = "", [CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0) {
+            if (mainThread.ThreadId != Thread.CurrentThread.ManagedThreadId) {
+                Debug.Fail(FormattableString.Invariant($"{memberName} at {sourceFilePath}:{sourceLineNumber} was incorrectly called from a background thread."));
```

Comment:
Should be in resource file?

Reply:
Don't think so. It is debug-only information. @MikhailArkhipov , what do you think?

Reply:
No, it is just debug string. Users won't see it. Besides, you don't want to see it in Chinese when you debug localized VS ;-)